### PR TITLE
Update contributing guide. Fix #1786

### LINKED ISSUE #1786: Add contributor information to developers wiki page
#1604 mentions that the developer documentation is outdated. Besides updating it, I think there is some information missing for contributors.

The main things I think are missing:
- How to set up a developer environment
- How to build Gensim
- How to send a pull request following the git flow described in the page

While the information provided seems appropriate for core developers of the project, contributors sending pull requests would in my opinion require those additional sections.

Regarding the developer environment, I don't see a requirements.txt (or requirements-dev.txt) file that can be used to install it. I guess there are also other project requirements for the project, and it'd be nice to know how contributors are expected to set up those things.

About the git flow, I think it should be obvious for open source contributors how to send a pull request to master. But given that the pull requests need to be sent to develop, I think it'd be useful to have a short documentation on how the whole flow looks like. It's nicely documented for core developers who can push to the repo, but not for contributors sending pull requests.

May be it could be a good idea to have separate pages for core developers and external contributors. And may be move this content to the CONTRIBUTING.md page.

### PR DESCRIPTION
**Pull request by menshikh-iv**
Fix #1786

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
 
 **The proper place for open-ended questions is the [gensim mailing list](https://groups.google.com/forum/#!forum/gensim).** Github is not the right place for research discussions or feature requests.
 
-For developers: see our [Developer Page](https://github.com/piskvorky/gensim/wiki/Developer-page#code-style) for details on code style, testing and similar.
+# How to add a new feature or create a pull request?
+
+1. <a href="https://github.com/RaRe-Technologies/gensim/fork">Fork the Gensim repository</a>
+2. Clone your fork: `git clone https://github.com/<USERNAME>/gensim.git`
+3. Create a new branch based on `develop`: `git checkout -b my-feature develop`
+4. Implement your changes
```
Comment (piskvorky) [inline CONTRIBUTING.md]: Python work environment? Dependencies?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
+5. Check that everything's OK in your branch:
+   - Check it for PEP8: `tox -e flake8`
+   - Build its documentation (works only for MacOS/Linux): `tox -e docs`
+   - Run unit tests: `tox -e py{version}-{os}`, for example `tox -e py27-linux` or `tox -e py36-win` where
+      - `{version}` is one of `27`, `35`, `36` and
+      - `{os}` is either `win` or `linux`
```
Comment (piskvorky) [inline CONTRIBUTING.md]: No testing for OSX?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,9 +1,29 @@
...
-For developers: see our [Developer Page](https://github.com/piskvorky/gensim/wiki/Developer-page#code-style) for details on code style, testing and similar.
+# How to add a new feature or create a pull request?
+
+1. <a href="https://github.com/RaRe-Technologies/gensim/fork">Fork the Gensim repository</a>
+2. Clone your fork: `git clone https://github.com/<USERNAME>/gensim.git`
+3. Create a new branch based on `develop`: `git checkout -b my-feature develop`
+4. Implement your changes
+5. Check that everything's OK in your branch:
+   - Check it for PEP8: `tox -e flake8`
+   - Build its documentation (works only for MacOS/Linux): `tox -e docs`
```
Comment (piskvorky) [inline CONTRIBUTING.md]: No docs for Windows?

Comment (menshikh-iv) [reply CONTRIBUTING.md]: No, never was.

File: CONTRIBUTING.md (hunk 2 above)
Comment (menshikh-iv) [reply CONTRIBUTING.md]: Probably, `linux` should works fine for macOS (but I can't test it)

File: CONTRIBUTING.md (hunk 1 above)
Comment (menshikh-iv) [reply CONTRIBUTING.md]: Fixed d6a956e

**Pull request merged**